controllers/storagecluster/cephconfig.go

```diff
@@ -151,5 +157,17 @@ func getRookCephConfig(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (s
 		}
 		return rookConfigData, nil
 	}
+	// additonal rook configs when it is a provider cluster
+	if sc.Spec.AllowRemoteStorageConsumers {
+		var rookConfigData string
+		var err error
+		for key, value := range extraRookConfigs {
+			rookConfigData, err = updateRookConfig(defaultRookConfigData, globalSectionKey, key, value)
```

Comment:
This will only apply one of the configs (the last one) as you are always using the `defaultRookConfigData` as your source

Also, why do we need to use  `updateRookConfig` if the config that we are trying to set is static and global?

Reply:
Ack. Fixed the issue